front-end/src/components/editProduct/EditProduct.js

```diff
@@ -0,0 +1,84 @@
+import { useHistory } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { productService } from '../../services/productServices';
+
+const Container = styled.div`
+    width: 80%;
+    min-height: 100vh;
+    background-color: whitesmoke;
+    margin: 0 auto 0 auto;
+
+    h2{
+        padding: 1rem;
+        text-align: center;
+    }
+    form{
+        display: flex;
+        flex-direction: column;
+        width: 35%;
+        margin: 0 auto 0 auto;
+    }
+
+    form input{
+        margin: 1rem;
+        padding: 0.8rem;
+        outline: none;
+        border: none;
+    }
+
+    form button {
+        padding: 0.8rem;
+        width: 25%;
+        margin: 2rem auto 0 auto;
+        cursor: pointer;
+        border: none;
+        background-color: white;
+        border-radius: 0.5rem;
+
+        :hover{
+            background-color: #BBB;
+            color: white;
+        }
+    }
+`;
+
+const AddProduct = (props) => {
+    const redirect = useHistory();
+    const productId = props.location.pathname.split('/')[2];
+    const [title, setTitle] = useState('');
+    const [category, setCategory] = useState('');
+    const [size, setSize] = useState('');
+    const [color, setColor] = useState('');
+    const [price, setPrice] = useState('');
+    const [imageUrl, setImageUrl] = useState('');
+
+    useEffect(() => {
+        productService.getSingeProduct(productId, setTitle, setCategory, setSize, setColor, setPrice, setImageUrl)
+    }, [productId])
+
+    const editProduct = () => {
+        productService.editProduct(productId, title, category, size, color, price, imageUrl)
+        redirect.push('/')
+    };
+
+
+    return (
+        <Container>
+            <h2>Add Product</h2>
+            <form>
+                <input type="text" value={title} onChange={(e) => setTitle(e.target.value)} placeholder="title" />
```

Comment:
You don't need to create arrow function. You can simplified it to this. And the `setTitle` function will receive the `event` and then extract `target.value` from there. Same for the other inputs.
```suggestion
                <input type="text" value={title} onChange={setTitle} placeholder="title" />
```

Reply:
I don't understand if I use only setTitle it returns an object.

Reply:
It returns you the event, and then you have to get the value in the function with event.target.value.

Reply:
So I should write another function and pass it. Not directly use the arrow function in onChange.

Reply:
I will show you tomorrow what I think. 😄

Reply:
Done something but not sure if it is right... I have extracted the value but it says that I'm changing from uncontrolled to controlled input. 100% I am not doing smth right.